src/App.tsx

```diff
@@ -8,7 +8,76 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SortType } from './types/FilterTypes';
+
+type ReorderOptions = {
+  filterType: SortType,
+  query: string,
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosLoaded, setTodosLoaded] = useState(true);
+
+  const [selectedTodo, setSelectedTodo] = useState<number | null>(null);
+
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState(SortType.All);
+
+  async function LoadTodos() {
+    const reponse = await getTodos();
+
+    setTodosLoaded(false);
+
+    setTodos(reponse);
+  }
+
+  useEffect(() => {
+    LoadTodos();
+  }, []);
+
+  const onChangeFilterType = (type: SortType) => {
+    setFilterType(type);
+  };
+
+  function getFilteredGoods({ filterType: type, query: input }: ReorderOptions) {
+    const filteredByFilter = [...todos].filter(todo => {
+      switch (type) {
+        case SortType.Active:
+          return todo.completed === false;
+
+        case SortType.Completed:
+          return todo.completed === true;
+
+        default:
+          return todo;
```

Comment:
```suggestion
          return true;
```
The filter's callback should return a boolean value. `todo` will be transformed to `boolean` automatically (always `true`), but it's better to do it explicitly.